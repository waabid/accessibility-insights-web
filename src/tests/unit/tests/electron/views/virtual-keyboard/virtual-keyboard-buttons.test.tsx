// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import {
    VirtualKeyboardButtons,
    VirtualKeyboardButtonsProps,
} from 'electron/views/virtual-keyboard/virtual-keyboard-buttons';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('VirtualKeyboardButtons', () => {
    let props: VirtualKeyboardButtonsProps;

    beforeEach(() => {
        props = {} as VirtualKeyboardButtonsProps;
    });

    test('renders with virtual keyboard collapsed', () => {
        props = {
            narrowModeStatus: {
                isVirtualKeyboardCollapsed: true,
            },
        } as VirtualKeyboardButtonsProps;
        const render = shallow(<VirtualKeyboardButtons {...props} />);

        expect(render.getElement()).toMatchSnapshot();
    });

    test('renders with virtual keyboard not collapsed', () => {
        props = {
            narrowModeStatus: {
                isVirtualKeyboardCollapsed: false,
            },
        } as VirtualKeyboardButtonsProps;
        const render = shallow(<VirtualKeyboardButtons {...props} />);

        expect(render.getElement()).toMatchSnapshot();
    });
});
