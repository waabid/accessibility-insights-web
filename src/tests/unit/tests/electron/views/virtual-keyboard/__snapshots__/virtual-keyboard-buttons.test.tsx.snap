// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualKeyboardButtons renders with virtual keyboard collapsed 1`] = `
<div
  className="virtualKeyboardButtons"
>
  <Button
    className="button"
    primary={true}
  >
    <span
      className="innerButtonContainer"
    >
      Tab
    </span>
  </Button>
  <Button
    className="button"
    primary={true}
  >
    <span
      className="innerButtonContainer"
    >
      <StyledIconBase
        iconName="upArrow"
      />
      Up
    </span>
  </Button>
  <Button
    className="button"
    primary={true}
  >
    <span
      className="innerButtonContainer"
    >
      <StyledIconBase
        className="left"
        iconName="upArrow"
      />
      Left
    </span>
  </Button>
  <Button
    className="button"
    primary={true}
  >
    <span
      className="innerButtonContainer"
    >
      <StyledIconBase
        className="down"
        iconName="upArrow"
      />
      Down
    </span>
  </Button>
  <Button
    className="button"
    primary={true}
  >
    <span
      className="innerButtonContainer"
    >
      <StyledIconBase
        className="right"
        iconName="upArrow"
      />
      Right
    </span>
  </Button>
  <Button
    className="button"
    primary={true}
  >
    <span
      className="innerButtonContainer"
    >
      Enter
    </span>
  </Button>
</div>
`;

exports[`VirtualKeyboardButtons renders with virtual keyboard not collapsed 1`] = `
<div
  className="virtualKeyboardButtons"
>
  <Button
    className="rectangleButton button"
    primary={true}
  >
    <span
      className="innerButtonContainer"
    >
      Tab
    </span>
  </Button>
  <div
    className="arrowButtonsContainer"
  >
    <Button
      className="button"
      primary={true}
    >
      <span
        className="innerButtonContainer"
      >
        <StyledIconBase
          iconName="upArrow"
        />
        Up
      </span>
    </Button>
    <div
      className="bottomRowButtons"
    >
      <Button
        className="button"
        primary={true}
      >
        <span
          className="innerButtonContainer"
        >
          <StyledIconBase
            className="left"
            iconName="upArrow"
          />
          Left
        </span>
      </Button>
      <Button
        className="button"
        primary={true}
      >
        <span
          className="innerButtonContainer"
        >
          <StyledIconBase
            className="down"
            iconName="upArrow"
          />
          Down
        </span>
      </Button>
      <Button
        className="button"
        primary={true}
      >
        <span
          className="innerButtonContainer"
        >
          <StyledIconBase
            className="right"
            iconName="upArrow"
          />
          Right
        </span>
      </Button>
    </div>
  </div>
  <Button
    className="rectangleButton button"
    primary={true}
  >
    <span
      className="innerButtonContainer"
    >
      Enter
    </span>
  </Button>
</div>
`;
